lib/spack/spack/directives.py

```diff
@@ -74,6 +75,15 @@ class OpenMpi(Package):
 directives = {}
 
 
+#
+# Types of dependencies.
```

Comment:
I think these are a better fit in `spack.spec`, since they're fundamental to how you consider the spec DAG (and they're used in query functions)